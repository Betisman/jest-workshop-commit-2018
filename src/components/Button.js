import React from 'react';

const Button = ({ className, children, onClick, value }) => 
  <button className={className} onClick={() => onClick(value)}>{children}</button>;

export default Button;
