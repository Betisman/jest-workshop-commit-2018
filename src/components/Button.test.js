import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from './Button';

const props = {
  children: 'Name button',
  onClick: () => 0,
  className: 'primary',
  value: 1,
};

describe('<Button />', () => {

  it('<Button /> match snapshot', () => {
    const tree = renderer.create(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  const wrapper = shallow(<Button {...props} />);
  it(`button contains ${props.className}`, () => {
    expect(wrapper.find(`.${props.className}`).exists()).toEqual(true);
  });

  it(`button contains ${props.children}`, () => {
    expect(wrapper.find(`.${props.className}`).props().children).toEqual(props.children);
  });

  // spys
  it('calls onClick prop when it is clicked', () => {
    const mockProps = {
      ...props,
      onClick: jest.fn(),
    };
    const newWrapper = shallow(<Button {...mockProps} />);
    newWrapper.find(`.${props.className}`).props().onClick();
    expect(mockProps.onClick).toHaveBeenCalledWith(props.value);
  });
});
