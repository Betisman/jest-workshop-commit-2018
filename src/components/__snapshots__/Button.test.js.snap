// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> <Button /> match snapshot 1`] = `
<button
  className="primary"
  onClick={[Function]}
>
  Name button
</button>
`;
